File: leetCode/fukun/lc_169.js
const majorityElement = function (nums) {
    let tmp = {};
    for (let i = 0; i < nums.length; i++) tmp[nums[i]] = tmp[nums[i]] ? ++tmp[nums[i]] : 1;
    for (v in tmp) if (tmp.hasOwnProperty(v) && tmp[v] > nums.length / 2) return v;
};

console.log(majorityElement([2, 2, 1, 1, 1, 2, 2]));

File: leetCode/XSS/lc_169.java
public class lc_169 {
    public static void main(String[] args) {

    }
    static class Solution {
        public int majorityElement(int[] nums) {
            int res=-1;int count=0;
            for(int i:nums)
            {
                if(res==i)
                {
                    count++;
                }
                else if(count>0)
                {
                    count--;
                }
                if(count==0)
                {
                    count++;
                    res=i;
                }
            }
            return res;
        }
    }
}
